### PR TITLE
fix/7830

### LINKED ISSUE #7830: Password Of the Day wrong timezone
**Describe the bug**
On password of the day rotation, the 2 date the code compare are not on the same timezone.

Per example:

```
Sep 26 15:26:10 packetfence pfperl-api-docker-wrapper[2531815]: pfperl-api(16) WARN: [mac:[undef]] NOW: 1695756369 (pf::pfcron::task::password_of_the_day::run)
Sep 26 15:26:10 packetfence pfperl-api-docker-wrapper[2531815]: pfperl-api(16) WARN: [mac:[undef]] Expiration: 1695798000 (pf::pfcron::task::password_of_the_day::run)
Sep 26 15:26:10 packetfence pfperl-api-docker-wrapper[2531815]: pfperl-api(16) WARN: [mac:[undef]] Expiration MYSQL: 2023-09-27T07:00:00 (pf::pfcron::task::password_of_the_day::run)
```

Now represent the time now which is "Tue Sep 26 2023 15:26:09 GMT-0400 (Eastern Daylight Time)"
Expiration represent the time that the password expire: "Wed Sep 27 2023 03:00:00 GMT-0400 (Eastern Daylight Time)"
And Expiration MYSQL represent the value we convert to epoch but it's converted to GMT+0

Normally Expiration value should be calculated on the local time zone and should be "Wed Sep 27 2023 07:00:00 GMT-0400 (Eastern Daylight Time)"

It could be a container image issue.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description

Use the database to calculate the expiration for POTD rotation.

# Issue
Fixes #7830

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Bug Fixes
* Use the database to calculate the expiration for POTD rotation.

### CODE REVIEW

File: t/unittest/pfcron/task.t [hunk 1]
```diff
@@ -114,7 +129,79 @@ sub startSMTP {
...
+    my ($ctx) = @_;
+    pipe(my $reader, my $writer);
+    my $pid = pf::AtFork::pf_fork();
+    BAIL_OUT("Cannot fork") if ( !defined $pid );
+    if ($pid) {
+        close($writer);
+        my $line = <$reader>;
+        close($reader);
+        #        pf::password::_delete('potd');
+        #pf::person::person_delete('potd');
```
Comment (satkunas) [inline t/unittest/pfcron/task.t]: Remove

**Comment by JeGoi** [comment]
Naive questions:
- Why using the DB timestamp instead of the PF config Time zone?
- What will happen in SaaS if DB is geo distributed?

**Comment by JeGoi** [comment]
@jrouzierinverse any update?
We may have a conflict with that: https://github.com/inverse-inc/packetfence/pull/7862/commits/931ab960f1614664f87f32b2ae31b1deedae9fcc
Thanks

**Comment by fdurand** [comment]
We fix the issue with https://github.com/inverse-inc/packetfence/pull/7862

**Pull request closed without merging**